homeassistant/components/intellifire/config_flow.py

```diff
@@ -4,60 +4,126 @@
 from typing import Any
 
 from aiohttp import ClientConnectionError
-from intellifire4py import IntellifireAsync
+from intellifire4py import AsyncUDPFireplaceFinder, IntellifireAsync
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.const import CONF_HOST
-from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResult
 
-from .const import DOMAIN
+from .const import DOMAIN, LOGGER
 
 STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
 
+MANUAL_ENTRY_STRING = "IP Address"  # Simplified so it does not have to be translated
 
-async def validate_input(hass: HomeAssistant, host: str) -> str:
+
+async def validate_host_input(host: str) -> str:
     """Validate the user input allows us to connect.
 
     Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
     """
     api = IntellifireAsync(host)
     await api.poll()
-
+    ret = api.data.serial
+    LOGGER.debug("Found a fireplace: %s", ret)
     # Return the serial number which will be used to calculate a unique ID for the device/sensors
-    return api.data.serial
+    return ret
 
 
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for IntelliFire."""
 
     VERSION = 1
 
-    async def async_step_user(
+    def __init__(self):
+        """Initialize the Config Flow Handler."""
+        self._config_context = {}
+        self._not_configured_hosts: list[str] = []
+
+    async def _find_fireplaces(self):
+        """Perform UDP discovery."""
+        fireplace_finder = AsyncUDPFireplaceFinder()
+        discovered_hosts = await fireplace_finder.search_fireplace(timeout=1)
+        configured_hosts = {
+            entry.data[CONF_HOST]
+            for entry in self._async_current_entries(include_ignore=False)
+            if CONF_HOST in entry.data  # CONF_HOST will be missing for ignored entries
+        }
+
+        self._not_configured_hosts = [
+            ip for ip in discovered_hosts if ip not in configured_hosts
+        ]
+        LOGGER.debug("Discovered Hosts: %s", str(discovered_hosts))
+        LOGGER.debug("Configured Hosts: %s", str(configured_hosts))
+        LOGGER.debug("Not Configured Hosts: %s", str(self._not_configured_hosts))
```

Comment:
```suggestion
        LOGGER.debug("Discovered Hosts: %s", discovered_hosts)
        LOGGER.debug("Configured Hosts: %s", configured_hosts)
        LOGGER.debug("Not Configured Hosts: %s", self._not_configured_hosts)
```

`%s` already asks for a string